### PR TITLE
Add permissions class to FeatureViewset, needed to access layer's fea…

### PR DESCRIPTION
**Pull request by nahuelange**
…tures

### CODE REVIEW

File: geostore/permissions.py [hunk 1]
```diff
@@ -32,3 +32,36 @@ def has_object_permission(self, request, view, obj):
...
+    def has_object_permission(self, request, view, obj):
+        if request.user.is_superuser:
+            return True
+
+        has_perm = request.user.has_perm('geostore.can_manage_layers')
+
+        if obj.layer.authorized_groups.exists():
+            # Checking permissions if the layer has group authorizations
+            is_authorized = any(map(
+                lambda group: group in obj.layer.authorized_groups.all(),
```
Comment (SebCorbin) [inline geostore/permissions.py]: I wound extract this queryset in a variable. Else the more the user has groups, the more queries are made.

File: geostore/permissions.py [hunk 2]
```diff
@@ -32,3 +32,36 @@ def has_object_permission(self, request, view, obj):
...
+                request.user.groups()
+            ))
+
+            if request.method not in SAFE_METHODS:
+                return is_authorized and has_perm
```
Comment (SebCorbin) [inline geostore/permissions.py]: An early return would prevent any queries if user has no permission
```python
if request.method not in SAFE_METHODS and not has_perm:
    return False
return any(map(
    lambda group: group in obj.layer.authorized_groups.all(),
    request.user.groups()
 ))
```

File: geostore/permissions.py [hunk 3]
```diff
@@ -32,3 +32,36 @@ def has_object_permission(self, request, view, obj):
...
+
+            return is_authorized
+
+        else:
+            # else we check normal privileges
+            if request.method in SAFE_METHODS:
+                return True
+            else:
+                return has_perm
```
Comment (SebCorbin) [inline geostore/permissions.py]:
```python
return request.method in SAFE_METHODS or has_perm
```
The main else is also optional

**Review by SebCorbin — CHANGES_REQUESTED** [review]

File: geostore/permissions.py (hunk 1 above)
Comment (nahuelange) [reply geostore/permissions.py]: fixed/improved

File: geostore/permissions.py (hunk 2 above)
Comment (nahuelange) [reply geostore/permissions.py]: fixed

File: geostore/permissions.py (hunk 3 above)
Comment (nahuelange) [reply geostore/permissions.py]: fixed

**Review by submarcos — APPROVED** [review]

**Pull request merged**